### ISSUE TITLE
tarball release: no version information (UNKNOWN)

### ISSUE THREAD

**Issue by LionNatsu** [traviscross/mtr#198]
(Tarball from https://github.com/traviscross/mtr/releases)
```
lion@Lion-Laptop [ mtr-0.92 ] $ ./bootstrap.sh 
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\${ <-- HERE ([^ \t=:+{}]+)}/ at /bin/automake line 3936.
configure.ac:8: installing 'build-aux/compile'
configure.ac:20: installing 'build-aux/config.guess'
configure.ac:20: installing 'build-aux/config.sub'
configure.ac:9: installing 'build-aux/install-sh'
configure.ac:9: installing 'build-aux/missing'
Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
lion@Lion-Laptop [ mtr-0.92 ] $ ./configure --version
mtr configure UNKNOWN
generated by GNU Autoconf 2.69

Copyright (C) 2012 Free Software Foundation, Inc.
This configure script is free software; the Free Software Foundation
gives unlimited permission to copy, distribute and modify it.
lion@Lion-Laptop [ mtr-0.92 ] $ 
```

It seems that mtr can now only build by git clone, but cannot by released tarball?

**Comment by rewolff** [comment]
```
git clone https://github.com/traviscross/mtr.git mtr4
cd mtr4
git log v0.92 | head -1 ; git log |head -1
commit 6cddfc6e8f0e20af4af218b545f83d43b3f9a387
commit 6cddfc6e8f0e20af4af218b545f83d43b3f9a387
./bootstrap.sh 
./configure --version
mtr configure 0.92

```
You seem to be using a version that is not the one tagged as 0.92.

**Comment by LionNatsu** [comment]
@rewolff No, I mean, you will see some "zip", "tar.gz" in https://github.com/traviscross/mtr/releases.
For example, https://github.com/traviscross/mtr/archive/v0.92.tar.gz .

It strips `.git` thus there is no any version control system.
But the fact is the script do depends on git.

I have an idea. When your script makes a release, write the version to certain file. Use that file to figure the version in building.

**Comment by rewolff** [comment]
OK.... So, for that to work, I would need to have the release-script put the version number somewhere, that the script bootstrap.sh can find it and use that.... 

The one I ship, on ftp://ftp.bitwizard.nl/mtr/ I think does things correctly. 



build-aux/git-version-gen says: 
# Then use the following lines in your Makefile.am, so that .version
# will be present for dependencies, and so that .version and
# .tarball-version will exist in distribution tarballs.

# EXTRA_DIST = $(top_srcdir)/.version
# BUILT_SOURCES = $(top_srcdir)/.version
# $(top_srcdir)/.version:
#       echo $(VERSION) > $@-t && mv $@-t $@
# dist-hook:
#       echo $(VERSION) > $(distdir)/.tarball-version

but it seems that this has not been done. Do you agree?

**Comment by LionNatsu** [comment]
Indeed. But even the one in FTP archive (ftp://ftp.bitwizard.nl/mtr/mtr-0.92.tar.gz), has something wrong.
```
lion@Lion-Laptop [ mtr-0.92 ] $ ./configure --version
mtr configure 0.91.1-4c982
```

Moreover, the README is confusing:
```
INSTALLING

  If you're building this from a tarball, compiling mtr should be as
  simple as:

	make

  It should first call the "configure" script and then run "make" again
  with the makefile that "configure" just generated. 

  If you're building from the git repository, you'll need to run:

	./bootstrap.sh && ./configure && make
```
I think it should be `./configure && make` when someone builds from tarball.

This maybe helpful for you to improve the make-a-release script to upload custom assets: https://developer.github.com/v3/repos/releases/#upload-a-release-asset
A little suggestion: you may use `curl`

**Comment by rewolff** [comment]
Thanks. Readme updated.

**Comment by anthraxx** [comment]
you could use some export-subst magic automatically to maintain a version file by git. This will also include and affect git archives generated from github itself.
It would be highly appreciated if the github release balls actually work as good as the custom tarballs on your FTP and export-subst is exactly what can solve this without the need of "complicated" release scripts.

https://stackoverflow.com/questions/19659976/add-tag-in-export-subst-file

**Comment by matt-kimball** [comment]
This is certainly a problem with the v0.92 release.

Surely the released tarball **should** contain a generated `configure` script and **should not** contain `bootstrap.sh`.  However, the v0.92 release **does not** contain a generated `configure` and **does** contain `bootstrap.sh`.

If the tarball shipped with a generated `configure` script, the version numbering would work as expected.  However, because it does not, the version built from the tarball is reported as `mtr UNKNOWN`.

I don't know the process @rewolff is using to generate the tarball, but when I do a `make release`, I get a tarball as expected -- with a generated `configure` and without `bootstrap.sh`.

**Comment by rewolff** [comment]
:-(
How do I "make release"? I get make: *** No rule to make target 'release'.  Stop.

**Comment by matt-kimball** [comment]
Sorry.

 I mean `make dist` not `make release`.

**Comment by rewolff** [comment]
Ok... .Tested, seems to work, patched the release script to use make dist. I also moved the "allow roger to interrupt the whole release" to before committing things to git. (I get to review the diff before that). 

This moves my release script back to testing.... :-)
